snagsby/sources.py

```diff
@@ -65,6 +60,59 @@ def options(self):
     def region_name(self):
         return self.options.get('region')
 
+    def get_data(self):
+        return sanitize(self.get_raw_data())
+
+
+class SMSource(AWSSource):
+    def get_sm_response(self):
+        key = "{}{}".format(self.url.netloc, self.url.path)
+        session = boto3.session.Session()
+        endpoint_url = "https://secretsmanager.{}.amazonaws.com".format(
+            self.region_name)
+        client = session.client(
+            service_name='secretsmanager',
```

Comment:
I happened to already have an older boto3 (1.5) in the venv I had around for this project and it turns out we do in fact need `boto3>=1.7` to use the `secretsmanager` client regardless of instantiating the session and client this way.

This also breaks if `self.region_name` is `None` as the endpoint_url will not be right. Given we need the the newer boto3 anyway I think we should just instantiate the client like we do with s3 and change our boto3 requirement to `>=1.7`:

```
client_opts = {}
if self.region_name:
    client_opts['region_name'] = self.region_name
client = boto3.client('secretsmanager', **client_opts)
```

This worked in testing.

Reply:
Updated